invisible_cities/io/pmaps_io_test.py

```diff
@@ -214,14 +244,46 @@ def test_load_pmaps(output_tmpdir, data):
         write = pmpio.pmap_writer(output_file)
         list(map(write, true_pmaps, event_numbers))
 
-    read_pmaps = pmpio.load_pmaps(pmap_filename)
+        write = reio.run_and_event_writer(output_file)
+        for event_number in event_numbers:
+            write(0, event_number, 0)
+
+    read_pmaps = pmpio.load_pmaps_eager(pmap_filename)
 
     assert len(read_pmaps) == len(true_pmaps)
     assert np.all(list(read_pmaps.keys()) == event_numbers)
     for true_pmap, read_pmap in zip(true_pmaps, read_pmaps.values()):
         assert_PMap_equality(read_pmap, true_pmap)
 
 
+def test_load_pmaps_lazy(KrMC_pmaps_filename):
+    """Ensure the lazy and non-lazy versions provide the same result"""
+    pmaps_eager = pmpio.load_pmaps_eager(KrMC_pmaps_filename)
+    pmaps_lazy  = pmpio.load_pmaps_lazy (KrMC_pmaps_filename)
```

Comment:
Should these not follow the new format shown in the test below: `load_pmaps(..., lazy=False/True)`

Reply:
I prefer to test the functions directly. A separate test ensures that `load_pmaps` (meant to be the interface to these functions(*)) works correctly with both values of the `lazy` parameter. But that's another layer. Testing the functions directly is more robust; in theory, something in `load_pmaps` could prevent errors in one of the inner functions from being spotted.

(*) Perhaps, we should "hide" the inner functions by prefixing an underscore to their names. This indicates the user that they should use `load_pmaps` with the corresponding value of `lazy` and not `load_pmaps_whatever`. What do you think?

Reply:
This is more of a Python convention than an IC one. Even though we don't follow PEP8 entirely, this specific rule is a common convention in the Python community. It's described [here](https://peps.python.org/pep-0008/#descriptive-naming-styles) (last paragraph).

Reply:
The takeaway message here is that using this convention is just a way of expressing something. That something is that users should exclusively use the `load_pmaps` function, and not the other ones. We don't forbid their use, just discourage it. If a user insists on calling the inner functions directly, they will have to import them explicitly. Unfortunately, there is no other solution for the case you mention than learning a bit more about the language.